### DISCUSSION TITLE
invalid key when chunks are formatted with non-zero only

### DISCUSSION THREAD

**Discussion by aleen42** [webpack/webpack#15264] [Q&A]
# Bug report

**What is the current behavior?**

Recently, I have found that when a chunk is not formatted with all numbers, it can't be loaded because of the undefined variable:

```js
/*! For license information please see 117.4b325.js.LICENSE.txt */
(self.webpackJsonp = self.webpackJsonp || []).push([[117], {
    11543: function(e, t, i) {
        var n = i(61493);
        i(f912e); // Throw error here
   },
   'f912e': function () {
   }
});
```

**Other relevant information:**
webpack version:  5.67.0
Node.js version: 17.3.1
Operating System: Windows OS
Additional tools:

**Comment by alexander-akait** [comment]
Can you provide configuration?

**Reply by vankop** [reply]
I think there was relative issue already..

**Reply by aleen42** [reply]
@vankop any relative information?

**Reply by aleen42** [reply]
I have tested and it seems that the breaking point is upgrading `terser-webpack-plugin` from `5.1.4` to `5.2.0`.

**Reply by aleen42** [reply]
It is weird to compare when passing `{output: {quote_style: 1}}` to the `terser`, resulting in the number of an asset has been changed into a hexadecimal format:

![image](https://user-images.githubusercontent.com/9573300/151495771-4e7c5a91-577f-405d-937f-2ca4433b60d1.png)

**Reply by aleen42** [reply]
Finally, it directly points to the problem of `webpack/lib/optimize/RealContentHashPlugin.js`, where the replacement of content of assets is not strictly, resulting in this problem. #15267

**Reply by vankop** [reply]
I take a look on the problem, for now I suggest to use `optimization.realContentHash=false`
